orb-ui/src/sound/mod.rs

```
@@ -388,39 +463,68 @@ impl Player for Fake {
         Ok(())
     }
 
+    /// Queue new sound.
+    ///
     /// Finds the sound file path for the given sound type and sends it to the
     /// sound player.
-    fn queue(&self, sound_type: Type) {
-        if let Some(sound_file) = self.sound_files.get(&sound_type) {
-            if let Some(sound_file) = sound_file.value() {
-                if let Ok(mut tx_queue) = self.queue_file.lock() {
-                    if let Err(err) = tx_queue.try_send(sound_file.clone()) {
-                        tracing::error!("Failed to queue sound: {:?}", err);
-                    }
-                }
-            } else {
-                tracing::error!(
-                    "Sound file {:?} doesn't have a known file path",
-                    sound_type
-                );
-            }
+    fn queue(&self, sound_type: Type) -> Result<()> {
+        let Some(sound_file) = self.sound_files.get(&sound_type) else {
+            eyre::bail!("Sound not found: {:?}", sound_type);
+        };
+
+        let Some(sound_file) = sound_file.value() else {
+            eyre::bail!("Sound {:?} doesn't have a known file path", sound_type);
+        };
+
+        if let Ok(mut tx_queue) = self.queue_file.lock() {
+            tx_queue
+                .try_send(sound_file.clone())
+                .wrap_err("Failed to queue sound")?;
         } else {
-            tracing::error!("Sound file not found: {:?}", sound_type);
+            eyre::bail!("Failed to lock queue")
```

instead of logging, or returning Err, we should panic if the lock is poisoned.